### PR TITLE
Feature/travis ci

### PR DESCRIPTION
**Pull request by piiertho**
This adds platform parameter for native modules, so that we only build the desired platform.
Example : `./gradlew :wrapper:godot-library:build -Pplatform=windows`

This adds Travis CI to automatically build commits and deploy tagged commits to bintray depot.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "windows" ]]; then
+    export JAVA_HOME=${JAVA_HOME:-/c/jdk}
+    && export PATH=${JAVA_HOME}/bin:${PATH}
+    && choco install jdk8 -params 'installdir=c:\\jdk' -y
+    && export GRADLE_OPTS=-Dorg.gradle.daemon=false;
+    fi
+install:
+  - ./gradlew :tools:api-classes-generator:build
+  - if [[ "$TRAVIS_OS_NAME" == "windows" ]]; then
```
Comment (chippmann) [inline .travis.yml]: Why do we have to execute this here again? It already should have happened at line `10-15` shouldn't it?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    else
+    ./gradlew :wrapper:godot-library:build -Pplatform=$TRAVIS_OS_NAME;
+    fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    ./gradlew :wrapper:godot-library-extension:build -Pplatform=macos;
+    else
+    ./gradlew :wrapper:godot-library-extension:build -Pplatform=$TRAVIS_OS_NAME;
+    fi
+  - ./gradlew :wrapper:godot-library:build -Pplatform=
+  - ./gradlew :wrapper:godot-library-extension:build
```
Comment (chippmann) [inline .travis.yml]: Why are these here without `platform` parameter value?
the builds should have happend already with lines `25, 27, 30, 32`?

File: .travis/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  ./gradlew :wrapper:godot-library-extension:bintrayUpload -PbintrayUser=$1 -PbintrayKey=$2 -Pplatform=metadata
+fi
+
+if [ $3 == "osx" ]; then
+  ./gradlew :wrapper:godot-library:bintrayUpload -PbintrayUser=$1 -PbintrayKey=$2 -Pplatform=macos
+  ./gradlew :wrapper:godot-library-extension:bintrayUpload -PbintrayUser=$1 -PbintrayKey=$2 -Pplatform=macos
+else
+  ./gradlew :wrapper:godot-library:bintrayUpload -PbintrayUser=$1 -PbintrayKey=$2 -Pplatform=$3
+  ./gradlew :wrapper:godot-library-extension:bintrayUpload -PbintrayUser=$1 -PbintrayKey=$2 -Pplatform=$3
+fi
```
Comment (chippmann) [inline .travis/deploy.sh]: personally i would find a switch case statement with the default case resulting in an error better. But If you prefer this: I'm fine with it.

File: tools/api-classes-generator/build.gradle.kts [hunk 4]
```diff
@@ -27,4 +41,30 @@ tasks.clean {
...
+            userOrg = "utopia-rise"
+            repo = "kotlin-godot"
+
+            name = project.name
+            vcsUrl = "https://github.com/utopia-rise/kotlin-godot-wrapper"
+            setLicenses("Apache-2.0")
+            version(closureOf<com.jfrog.bintray.gradle.BintrayExtension.VersionConfig> {
+                this.name = project.version.toString()
+                released = Date().toString()
+                description = "Godot gradle plugin ${project.version}"
```
Comment (chippmann) [inline tools/api-classes-generator/build.gradle.kts]: Shouldn't be the description "Api Classes Generator ${project.version}" ?

File: tools/entry-generator/build.gradle.kts [hunk 5]
```diff
@@ -27,12 +30,38 @@ application {
...
+            userOrg = "utopia-rise"
+            repo = "kotlin-godot"
+
+            name = project.name
+            vcsUrl = "https://github.com/utopia-rise/kotlin-godot-wrapper"
+            setLicenses("Apache-2.0")
+            version(closureOf<com.jfrog.bintray.gradle.BintrayExtension.VersionConfig> {
+                this.name = project.version.toString()
+                released = Date().toString()
+                description = "Godot gradle plugin ${project.version}"
```
Comment (chippmann) [inline tools/entry-generator/build.gradle.kts]: Shouldn't be the description "Entry Generator ${project.version}" ?

File: wrapper/godot-library-extension/build.gradle.kts [hunk 6]
```diff
@@ -16,25 +23,47 @@ kotlin {
...
+            userOrg = "utopia-rise"
+            repo = "kotlin-godot"
+
+            name = "${project.name}-$platform"
+            vcsUrl = "https://github.com/utopia-rise/kotlin-godot-wrapper"
+            setLicenses("Apache-2.0")
+            version(closureOf<com.jfrog.bintray.gradle.BintrayExtension.VersionConfig> {
+                this.name = project.version.toString()
+                released = Date().toString()
+                description = "Godot library ${project.version}"
```
Comment (chippmann) [inline wrapper/godot-library-extension/build.gradle.kts]: Shouldn't be the description "Godot Library Extension ${project.version}" ?

File: .travis.yml (hunk 2 above)
Comment (piiertho) [reply .travis.yml]: I Forgot to remove the lines, I'll fix this !

File: tools/api-classes-generator/build.gradle.kts (hunk 4 above)
Comment (piiertho) [reply tools/api-classes-generator/build.gradle.kts]: Yes !

File: .travis/deploy.sh (hunk 3 above)
Comment (piiertho) [reply .travis/deploy.sh]: Indeed a switch case statement would be better. But for now release is working. Maybe it would be better to rework it when we'll add android and ios support.

File: wrapper/godot-library-extension/build.gradle.kts (hunk 6 above)
Comment (piiertho) [reply wrapper/godot-library-extension/build.gradle.kts]: Yes !

File: .travis.yml (hunk 1 above)
Comment (piiertho) [reply .travis.yml]: Wrong copy paste :P
